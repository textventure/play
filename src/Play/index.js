import React, { Component } from 'react';
import Branch from '../Branch';
import { getKey, getValue } from '../helpers/util';

export default class Play extends Component {
  render() {
    const { branches, config, id } = this.props;
    const branch = branches[id];
    const choices = getValue(branch);
    const text = getKey(branch);

    return (
      <main>
        <Branch choices={choices} config={config} id={id} text={text} />
      </main>
    );
  }
}
