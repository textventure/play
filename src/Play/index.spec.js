import React from 'react';
import { shallow } from 'enzyme';
import Play from '.';

let props;
let wrapper;

describe('with props', () => {
  beforeAll(() => {
    props = {
      branches: {
        start: {
          text: ['choice1', 'choice2'],
        },
      },
      config: {},
      id: 'start',
    };
    wrapper = shallow(<Play {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
