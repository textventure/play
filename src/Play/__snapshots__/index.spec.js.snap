// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with props renders correctly 1`] = `
<main>
  <WithStyles(Branch)
    choices={
      Array [
        "choice1",
        "choice2",
      ]
    }
    config={Object {}}
    id="start"
    text="text"
  />
</main>
`;
